### ISSUE TITLE
Plugin reports intermittent errors/exceptions

### ISSUE THREAD

**Issue by linqcan** [kodi-svtplay/xbmc-svtplay#198]
Example of stack trace
```
 ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--- NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
Error Type: <class 'requests.exceptions.SSLError'>
Error Contents: ('bad handshake: SysCallError(0, None)',)
...
File "/home/johndoe/.kodi/addons/plugin.video.svtplay/resources/lib/svt.py", line 369, in __get_json
  response = requests.get(url)
File "/home/johndoe/.kodi/addons/script.module.requests/lib/requests/api.py", line 70, in get
  return request('get', url, params=params, **kwargs)
File "/home/johndoe/.kodi/addons/script.module.requests/lib/requests/api.py", line 56, in request
  return session.request(method=method, url=url, **kwargs)
File "/home/johndoe/.kodi/addons/script.module.requests/lib/requests/sessions.py", line 488, in request
  resp = self.send(prep, **send_kwargs)
File "/home/johndoe/.kodi/addons/script.module.requests/lib/requests/sessions.py", line 630, in send
  history = [resp for resp in gen] if allow_redirects else []
File "/home/johndoe/.kodi/addons/script.module.requests/lib/requests/sessions.py", line 190, in resolve_redirects
  **adapter_kwargs
File "/home/johndoe/.kodi/addons/script.module.requests/lib/requests/sessions.py", line 609, in send
  r = adapter.send(request, **kwargs)
File "/home/johndoe/.kodi/addons/script.module.requests/lib/requests/adapters.py", line 497, in send
  raise SSLError(e, request=request)

```

or

```
SSLError: hostname 'www.svtplay.se' doesn't match either of '*.svt.se', 'svt.se', '*.stage.svt.se', 'stage.svt.se', '*.app.svt.se', 'app.svt.se'
```

SVTPlay uses HTTPS/SSL only nowadays.

Probably related to this: https://github.com/kennethreitz/requests/issues/2022
TL;DR The OpenSSL version used in Kodi is suboptimal for SSL connections. Not plugin related.

**Known remedy/workaround:** Exit the plugin and try again. Usually works after a couple of tries.

**Comment by linqcan** [comment]
Will check with Kodi if they have any remedy or planned changes related to this

**Comment by even-stephen** [comment]
For me the problem only manifests itself when running on certain linux distros (a non-issue on windows or libreelec/openelec)
The following fixes the problem (on ubuntu 16.04)
sudo apt install libssl-dev
sudo apt install python-pip
sudo pip install ndg-httpsclient pyasn1 --upgrade

**Comment by linqcan** [comment]
Yes, it's a Kodi issue on certain distros has Kodi uses system Python and
that tend to be very different. I am trying to find out if this is
acknowledged by Kodi so I can refer users to them.

**Comment by linqcan** [comment]
Possible plugin change due to this; catch SSLError on UI level and display some kind of message instead of the "cryptic" fail/error that is seen now.

**Comment by linqcan** [comment]
After a clean install of Kodi profile and plugin from repo I see this, and Kodi dies unexpectedly:
```
22:34:48.460 T:139751200876288   ERROR: Previous line repeats 6 times.
22:34:48.461 T:139751200876288   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <class 'requests.exceptions.SSLError'>
                                            Error Contents: ('bad handshake: SysCallError(0, None)',)
                                            Traceback (most recent call last):
                                              File "/home/erik/.kodi/addons/plugin.video.svtplay/default.py", line 331, in <module>
                                                viewSection(ARG_MODE, int(ARG_PAGE))
                                              File "/home/erik/.kodi/addons/plugin.video.svtplay/default.py", line 102, in viewSection
                                                (items, more_items) = svt.getItems(section, page)
                                              File "/home/erik/.kodi/addons/plugin.video.svtplay/resources/lib/svt.py", line 319, in getItems
                                                json_data = __get_json(section_name+"?page="+str(page))
                                              File "/home/erik/.kodi/addons/plugin.video.svtplay/resources/lib/svt.py", line 392, in __get_json
                                                response = requests.get(url)
                                              File "/home/erik/.kodi/addons/script.module.requests/lib/requests/api.py", line 70, in get
                                                return request('get', url, params=params, **kwargs)
                                              File "/home/erik/.kodi/addons/script.module.requests/lib/requests/api.py", line 56, in request
                                                return session.request(method=method, url=url, **kwargs)
                                              File "/home/erik/.kodi/addons/script.module.requests/lib/requests/sessions.py", line 488, in request
                                                resp = self.send(prep, **send_kwargs)
                                              File "/home/erik/.kodi/addons/script.module.requests/lib/requests/sessions.py", line 609, in send
                                                r = adapter.send(request, **kwargs)
                                              File "/home/erik/.kodi/addons/script.module.requests/lib/requests/adapters.py", line 497, in send
                                                raise SSLError(e, request=request)
                                            SSLError: ('bad handshake: SysCallError(0, None)',)
                                            -->End of Python script error report<--
22:34:48.509 T:139752316598656   ERROR: GetDirectory - Error getting plugin://plugin.video.svtplay/?mode=popular
22:34:48.509 T:139752316598656   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.svtplay/?mode=popular) failed

```

**Comment by linqcan** [comment]
Removing "scripts.moodule.requests" solved this (system `requests` is used instead). Need to contact Kodi about this...

**Comment by linqcan** [comment]
Haven't seen this in a while, and most code is rewrtitten. Closing. Will open new issue if it occurs again.
